src/oas3/service.ts

```diff
@@ -38,11 +39,12 @@ export const transformOas3Service: Oas3HttpServiceTransformer = ({ document }) =
     httpService.termsOfService = document.info.termsOfService;
   }
 
-  if (document.info?.['x-logo']) {
+  if (document.info?.['x-logo'] && isXLogo(document.info['x-logo'])) {
```

Comment:
You can create a shared transformer and place it inside of `oas` directory.

```suggestion
  if (isDictionary(document.info?.['x-logo'])) {
     httpService.logo = translateLogo(document);
  }
```